core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -802,7 +807,12 @@ private[spark] class BlockManager(
       logDebug("Put block %s locally took %s".format(blockId, Utils.getUsedTimeMs(startTimeMs)))
       if (level.replication > 1) {
         // Wait for asynchronous replication to finish
-        Await.ready(replicationFuture, Duration.Inf)
+        try {
+          Await.ready(replicationFuture, Duration.Inf)
```

Comment:
@ScrapCodes, towards your other comment, I think that timeouts in this case might already happen to be covered by network / RPC timeouts within the `replicationFuture`'s code.